cmd/bedValueWig/bedValueWig.go

```diff
@@ -27,54 +32,68 @@ func bedMaxWig(infile string, database string, chromsizeFile string, outfile str
 			var wigCounterByChrom float64 = 0
 			for k := range wigData[i].Values { //Cycle through each value in the float64[]
 				var chromValueMultiplyByStep float64 = 0
-				chromValueMultiplyByStep = (float64(wigData[i].Step) * wigData[i].Values[k]) // multiply each value by the step for that chrom
+				chromValueMultiplyByStep = float64(wigData[i].Step) * wigData[i].Values[k] // multiply each value by the step for that chrom
 				wigCounterByChrom = wigCounterByChrom + chromValueMultiplyByStep
 			}
 			wigTotal += wigCounterByChrom
 		}
 	}
 
 	for i = 0; i < len(sizes); i++ {
-		chromSlice = WigChromToSlice(wigData, sizes[i].Size, sizes[i].Name)
+		chromIndex = getWigIndex(wigData, sizes[i].Name)
 		for k := 0; k < len(records); k++ {
 			if records[k].Chrom == sizes[i].Name {
 				currentBed = records[k]
 				if currentBed.FieldsInitialized < 7 {
 					currentBed.FieldsInitialized = 7
 				}
-				maxWig := bedRangeMax(chromSlice, records[k].ChromStart, records[k].ChromEnd)
+				if bedMinWigFlag {
+					currValue = bedRangeMin(wigData[chromIndex].Values, records[k].ChromStart, records[k].ChromEnd)
+				} else if bedAverageWigFlag {
+					currValue = bedRangeAverage(wigData[chromIndex].Values, records[k].ChromStart, records[k].ChromEnd)
+				} else {
+					currValue = bedRangeMax(wigData[chromIndex].Values, records[k].ChromStart, records[k].ChromEnd)
+				}
 				if norm == true {
-					maxWig = maxWig / wigTotal
+					currValue = currValue / wigTotal
 				}
-				currentBed.Annotation = append(currentBed.Annotation, fmt.Sprintf("%g", float64(maxWig))) // %g will
+				currentBed.Annotation = append(currentBed.Annotation, fmt.Sprintf("%g", currValue)) // %g will
 				// print %e for large exponents, %f otherwise.
 				// Precision for %g; it is the smallest number of digits necessary to identify the value uniquely.
-				outlist = append(outlist, currentBed)
+				outList = append(outList, currentBed)
 			}
 		}
 	}
-	bed.Write(outfile, outlist)
+	bed.Write(outfile, outList)
 }
 
-func WigChromToSlice(w []wig.Wig, size int, chrom string) []float64 {
-	output := make([]float64, size)
-	for _, v := range w {
+func getWigIndex(w []wig.Wig, chrom string) int {
+	for i, v := range w {
 		if v.Chrom == chrom {
-			output = v.Values
+			return i
 		}
 	}
-	return output
+	log.Fatalf("Chromosome with name: %v not found in wig.", chrom)
+	return -1
 }
 
-func sliceRangeAverage(w []float64, start int, end int) float64 {
+func bedRangeAverage(w []float64, start int, end int) float64 {
 	length := end - start
 	var sum float64
 	var i int
 
 	for i = 0; i < length; i++ {
 		sum = sum + w[i+start]
 	}
-	return (sum / float64(length))
+	return sum / float64(length)
+}
+
+func bedRangeMin(w []float64, start int, end int) float64 {
+	var min = w[start]
+	for i := start; i < end; i++ {
+		min = numbers.MinFloat64(min, w[i])
```

Comment:
This is probably an esoteric point that no one cares about but me, so I give you license to ignore me, but you might consider avoiding importing numbers when possible. I think min/max functions probably account for >90% of numbers imports. That would normally be fine, but the numbers package is probably one of the largest packages in gonomics so when you use the min/max from numbers, you also get MCMC, monte carlo models, and log arithmetic included in your binary. The current binary for gonomics is just under 600MB. It might be worth considering ways to trim that down.

Reply:
this feels related to our choice to pull the min/max functions out of the lightweight 'common' package. One option is we write a bunch of small min/max helper functions without our programs, the other is we think about a more appropriate place for those functions in gonomics. Numbers is a very broad package and could also be broken

Reply:
Breaking up numbers may be a good idea. I think that this will become less of an issue when generics are implemented (december release??) but for now is something worth considering.